src/Admin/Services/AccessControlService.cs

```diff
@@ -17,6 +21,18 @@ public class AccessControlService : IAccessControlService
         _globalSettings = globalSettings;
     }
 
+    public bool UserHasPermission(Permission permission)
+    {
+        if (_globalSettings.SelfHosted)
+            return true;
+
+        var userRole = GetUserRoleFromClaim();
```

Comment:
Is the assumption here that a user can only be in one role at a time?

Reply:
Yes! The requirement is for a user to only ever have one role.